components/buttons.tsx

```diff
@@ -0,0 +1,42 @@
+import type { FC } from "react";
+import React, { type ReactElement } from "react";
+import { signInUserWithGoogle, signOutUser } from "@/firebase/authentication";
+import Button from "@mui/material/Button";
+
+const handleAsyncClick = (asyncFunction: () => Promise<void>) => (): void => {
+  asyncFunction().catch((error: unknown) => {
```

Comment:
Use explicit `try/catch`, not method chaining, and definitely don't use `unknown` or `any` because it's not unknown... It has a very specific type

Reply:
Has been redone to not be as ugly and without the "unknown" type